homeassistant/components/rainmachine/switch.py

```diff
@@ -99,17 +100,27 @@
     """Set up RainMachine switches based on a config entry."""
     rainmachine = hass.data[RAINMACHINE_DOMAIN][DATA_CLIENT][entry.entry_id]
 
-    entities = []
-
-    programs = await rainmachine.client.programs.all(include_inactive=True)
-    for program in programs:
-        entities.append(RainMachineProgram(rainmachine, program))
+    tasks = {
+        "programs": rainmachine.client.programs.all(include_inactive=True),
+        "zones": rainmachine.client.zones.all(include_inactive=True),
+    }
 
-    zones = await rainmachine.client.zones.all(include_inactive=True)
-    for zone in zones:
-        entities.append(
-            RainMachineZone(rainmachine, zone, rainmachine.default_zone_runtime)
-        )
+    entities = []
+    results = await asyncio.gather(*tasks.values(), return_exceptions=True)
+    for kind, result in zip(tasks, results):
+        if isinstance(result, RequestError):
```

Comment:
Can there be another type of error? Do we need to check for that too?

Reply:
Great point. Here (and throughout `switch.py`), I've replaced `RequestError` with `RainMachineError`, which is a more general error that should catch everything.